### PR TITLE
[CAMEL-10779] Authentication : provide a way to use refresh_token mode in addition to password method

### PR DESCRIPTION
**Pull request by ajaira**
Objective is to enrich the SalesforceSession.getLoginRequest(HttpConversation) method and add 
fields.put("grant_type", "refresh_token");
as an alternative to 
fields.put("grant_type", "password");

jira: https://issues.apache.org/jira/browse/CAMEL-10779

### CODE REVIEW

**Comment by davsclaus** [comment]
@zregvart I wonder if you have time to help review this PR?

**Comment by zregvart** [comment]
:+1: @ajaira thanks for the contribution!

@davsclaus I would update the documentation and replace `== null` with `ObjectHelper::isEmpty`, but other than that it looks good to me, I've run tests with username/password and refresh token to verify.

File: components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/SalesforceSession.java [hunk 1]
```diff
@@ -72,8 +72,13 @@ public SalesforceSession(SalesforceHttpClient httpClient, long timeout, Salesfor
         ObjectHelper.notNull(config.getLoginUrl(), "loginUrl");
         ObjectHelper.notNull(config.getClientId(), "clientId");
         ObjectHelper.notNull(config.getClientSecret(), "clientSecret");
-        ObjectHelper.notNull(config.getUserName(), "userName");
-        ObjectHelper.notNull(config.getPassword(), "password");
+
+        if (config.getRefreshToken() == null) {
```
Comment (zregvart) [inline components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/SalesforceSession.java]: Could we use `ObjectHelper.isEmpty(config.getRefreshToken()` here?

File: platforms/spring-boot/components-starter/camel-salesforce-starter/src/main/java/org/apache/camel/component/salesforce/springboot/SalesforceComponentConfiguration.java [hunk 2]
```diff
@@ -832,6 +832,10 @@ public void setApprovalSkipEntryCriteria(
          */
         private String clientSecret;
         /**
+         * Salesforce connected application Consumer token
```
Comment (zregvart) [inline platforms/spring-boot/components-starter/camel-salesforce-starter/src/main/java/org/apache/camel/component/salesforce/springboot/SalesforceComponentConfiguration.java]: To keep in in line could this be `Salesforce connected application OAuth refresh token`?

**Comment by ajaira** [comment]
Hello Zoran,
I took into account your comments.

Thanks
Ajaira

**Comment by zregvart** [comment]
@ajaira Thanks :+1: 
@davsclaus this can be merged

**Comment by davsclaus** [comment]
Thanks guys will get this merge within the hour, have a built running now first

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Comment by mabroukb** [comment]
Thanks guys for merging our PR.
the Jira is also closed.

**Pull request closed without merging**